documentation/ntfs/ntfs_fare_extention.md

```diff
@@ -0,0 +1,69 @@
+ticket_use_perimeters.txt | Required | Defines the networks and lines of a ticket_use
+ticket_use_restrictions.txt | Optionnal | Defines a use of a ticket between zones or stop_areas
+
+## Other modifications
+The `object_properties.txt` and `object_codes.txt` files are extended to enable the reference of a ticket_id.
+
+# Added files descriptions
+
+## tickets.txt (Required)
+Field | Type | Constraint | Description
+--- | --- | --- | ---
+ticket_id | Text | Required | Unique ID of the ticket
+ticket_name | Text | Required | Name of the ticket (may be displayed to passengers)
+ticket_comment | Text | Optionnal | Comment on the ticket that may be displayed to the passenger
+
+## ticket_prices.txt (Required)
+Field | Type | Constraint | Description
+--- | --- | --- | ---
+ticket_id | Text | Required | Unique ID of the ticket
+ticket_price | Non-negative Float | Required |
+ticker_currency | Currency Code | Required | An ISO 4217 alphabetical currency code. For the list of current currency, refer to https://en.wikipedia.org/wiki/ISO_4217#Active_codes.
+ticket_validity_start | Date | Required | First day of usability of the ticket
+ticket_validity_end | Date | Required | Last day of usability of the ticket
+
+## ticket_uses.txt (Required)
+Field | Type | Constraint | Description
+--- | --- | --- | ---
+ticket_id | Text | Required | Unique ID of the ticket
+ticket_use_id | Text | Required | Unique ID of the ticket use
+max_transfers | integer | Required | Max number of transfers using one ticket. The value can be empty to indicate there is no constraint, but the field must be provided.
+boarding_time_limit | integer | Required | Specifies the time frame (in minutes) in which a passenger can board in a new vehicle. The value can be empty to indicate there is no constraint, but the field must be provided.
+alighting_time_limit | integer | Required | Specifies the time frame (in minutes) in which a passenger must get of the vehicle. When this duration is expired, the ticket is no longer valid. The value can be empty to indicate there is no constraint, but the field must be provided.
+
+## ticket_use_perimeter.txt (Required)
+
+A ticket_use must contain at least one included line or network.
+
+Field | Type | Constraint | Description
+--- | --- | --- | ---
+ticket_use_id | Text | Required | Unique ID of the ticket use
+object_type | Text | Required | This field can be eiher `line` or `network`
+object_id | Text | Required | This field refers to the ID of the considered line or the network.
+perimeter_action | integer | Required | Value 1 for inclusion (for a `network` or a `line`) and 2 for an exclusion (for a `line` only)
+
+## ticket_use_restriction.txt (Optionnal)
+Field | Type | Constraint | Description
+--- | --- | --- | ---
+ticket_use_id | Text | Required | Unique ID of the ticket use
+restriction_type | Text | Required | either `zone` to specify a restriction based on `fare_zone_id` of stops or `OD` for a stop_area id use.
```

Comment:
I don't like the zones to be a min and a max, as zones can be letters or numbers (or any string) and we would then rank them alphabetically to compare to min:max (how about 1,2,11).
I'd rather have a list of zones.

Will it be useful at some point to allow/forbid passing by other zones ? Or can we just always check OD of sections and it should be OK ?

Reply:
In the proposed modelisation, a fare will be provided for a start within the `1` zone and an end in the `11` zone. If you want to go from zone `1` to `2`, it can be specified as an other ticket, with a different price (or the same ticket, with an other ticket_use)